mlflow/utils/databricks_utils.py

```diff
@@ -497,37 +532,66 @@ def get_databricks_host_creds(server_uri=None):
 
     Returns:
-        MlflowHostCreds which includes the hostname and authentication information necessary to
+        MlflowHostCreds which includes the constructed Databricks SDK workspace client instance,
+        or the hostname and authentication information necessary to
         talk to the Databricks server.
     """
-    profile, path = get_db_info_from_uri(server_uri)
-    config = ProfileConfigProvider(profile).get_config() if profile else get_config()
-    # if a path is specified, that implies a Databricks tracking URI of the form:
-    # databricks://profile-name/path-specifier
-    if (not config or not config.host) and path:
-        dbutils = _get_dbutils()
-        if dbutils:
-            # Prefix differentiates users and is provided as path information in the URI
-            key_prefix = path
-            host = dbutils.secrets.get(scope=profile, key=key_prefix + "-host")
-            token = dbutils.secrets.get(scope=profile, key=key_prefix + "-token")
-            if host and token:
-                config = DatabricksConfig.from_token(host=host, token=token, insecure=False)
-    if not config or not config.host:
-        _fail_malformed_databricks_auth(profile)
 
-    insecure = hasattr(config, "insecure") and config.insecure
+    from databricks.sdk import WorkspaceClient
 
-    if config.username is not None and config.password is not None:
-        return MlflowHostCreds(
-            config.host,
-            username=config.username,
-            password=config.password,
-            ignore_tls_verification=insecure,
-        )
-    elif config.token:
-        return MlflowHostCreds(config.host, token=config.token, ignore_tls_verification=insecure)
-    _fail_malformed_databricks_auth(profile)
+    profile, _ = get_db_info_from_uri(server_uri)
+    try:
+        # Using databricks-sdk to create Databricks WorkspaceClient instance,
+        # If authentication is failed, exception is raised.
+        # databricks-sdk supports many kinds of authentication ways,
+        # it will try to read authentication information by the following ways:
+        # 1. Read dynamic generated token via databricks `dbutils`.
+        # 2. parse relevant environment variables (such as DATABRICKS_HOST + DATABRICKS_TOKEN
+        #    or DATABRICKS_HOST + DATABRICKS_CLIENT_ID + DATABRICKS_CLIENT_SECRET)
+        #    to get authentication information
+        # 3. parse ~/.databrickscfg file (generated by databricks-CLI command-line tool)
+        #    to get authentication information.
+        # databricks-sdk is designed to hide authentication details and
+        # support various authentication ways, so that it does not provide API
+        # to get credential values. Instead, we can use ``WorkspaceClient``
+        # API to invoke databricks shard restful APIs.
+        ws_client = WorkspaceClient(profile=profile)
+        ws_client.config.retry_timeout_seconds = \
+            MLFLOW_DATABRICKS_ENDPOINT_HTTP_RETRY_TIMEOUT.get()
```

Comment:
This doesn't look like the right way to set options. See https://github.com/databricks/databricks-sdk-py?tab=readme-ov-file#additional-authentication-configuration-options